### PR TITLE
docs(progress-bar) : Textual improvements for Buffer & Query modes

### LINKED ISSUE #19817: docs-bug(COMPONENT): Textual improvements of the progress-bar documentation
#### Documentation Feedback

In the documentation on progress bars, a few times the description says "... where the user wants to indicate ..." but this is incorrect. The user does not want to indicate anything here. Also, the texts are gramatically incorrect and confusing.

https://material.angular.io/components/progress-bar/overview#buffer

- Current text: "Operations where the user wants to indicate some activity or loading from the server, use the buffer indicator."
- Suggestion: "Use the buffer indicator to indicate loading from the server in addition to the actual progress."

https://material.angular.io/components/progress-bar/overview#query

- Current text: "For situations where the user wants to indicate pre-loading (until the loading can actually be made), use the query indicator."
- Suggestion: "Use the query indicator to indicate pre-loading before the actual loading starts."

Thanks,
Peter

### PR DESCRIPTION
**Pull request by walvekarnikhil**
Fixes #19817

### CODE REVIEW

File: src/material/progress-bar/progress-bar.md [hunk 1]
```diff
@@ -20,17 +20,16 @@ not necessary to indicate how long it will take should use the indeterminate ind
 In this mode the `value` property is ignored.
 
 #### Buffer
-Operations where the user wants to indicate some activity or loading from the server, 
-use the buffer indicator.
+Use the buffer indicator when want to indicate some activity or loading from the server.
```
Comment (crisbeto) [inline src/material/progress-bar/progress-bar.md:23]: "when want" -> "when you want"

File: src/material/progress-bar/progress-bar.md [hunk 2]
```diff
@@ -20,17 +20,16 @@ not necessary to indicate how long it will take should use the indeterminate ind
...
 
 <!-- example(progress-bar-buffer) -->
 
 In "buffer" mode, `value` determines the progress of the primary bar while the `bufferValue` is 
 used to show the additional buffering progress.
 
 #### Query
-For situations where the user wants to indicate pre-loading (until the loading can actually be made), 
-use the query indicator.
+Use the query indicator to indicate pre-loading before the actual loading starts.
```
Comment (crisbeto) [inline src/material/progress-bar/progress-bar.md:31]: "Indicator to indicate" sounds a little weird. Maybe "Use the `query` mode of the progress bar to indicate..."?

File: src/material/progress-bar/progress-bar.md (hunk 1 above)
Comment (crisbeto) [inline src/material/progress-bar/progress-bar.md:23]: "Indicator to indicate" sounds weird, maybe "Use the `buffer` mode of the progress bar..."?

File: src/material/progress-bar/progress-bar.md (hunk 2 above)
Comment (walvekarnikhil) [reply src/material/progress-bar/progress-bar.md:31]: This sounds way better :)

**Pull request merged**